src/client/components/pages/Courses/CourseInstanceTable.tsx

```diff
@@ -323,4 +323,16 @@ const CourseInstanceTable: FunctionComponent<CourseInstanceTableProps> = ({
   );
 };
 
-export default CourseInstanceTable;
+/**
+ * Memoize the redered content of the CourseInstanceTable, and only update if
+ * the actual list of courses, the set of columnns or the filters change
+ * (returning true from the second argument skips updating).
+ */
+export default React.memo(
+  CourseInstanceTable,
+  (prev, next) => (
+    prev.tableData === next.tableData
+  && prev.filters === next.filters
+  && prev.courseList === next.courseList
+  )
+);
```

Comment:
I was seeing performance issues on [my branch](https://github.com/seas-computing/course-planner/tree/feature/424-course-text-filters), which adds text filters to the `CourseInstanceTable.` When I typed in the filter field, this updated the `filters` state, causing the entire table needing to be re-rendered in between key strokes. This caused a very noticeable delay between typing and the filter field displaying what was typed. As a workaround, I separated the Course Instance Body from the rest of the table and memoized the `CourseInstanceTableBody` instead of the entire table (commit 959362fc39e76bb927c668b58a9be3a084ed558b) so that typing in the text filter field lets the user see immediately what was typed. Maybe we can take a look at this again together next week.

Reply:
I think that's a good approach. In fact, I think it might be worth going a step farther and also splitting out the `TableHead` section, which would let us be more specific about which props are passed into the separate components and which are memoized (i.e., we don't need to update the table head unless the filters/columns have changed). 

I agree that we should look at this together though, so we can try to sort out how to actually make those changes without getting into a huge merge conflict mess.

Reply:
Sounds good! We can work through this tomorrow.